GoogleForCreators/web-stories-wp
# Pre-Publish Checklist: Add expand 'hidden page' functionality

## Context

#9350 Feature request

## Summary

Adds functionality to toggle whether page previews are shown in the checklist.

## Relevant Technical Choices



## To-do

- [x] Add tests somewhere

## User-facing changes

<img src="https://user-images.githubusercontent.com/22185279/147490474-2413c06d-f588-4db1-95fa-510cb469da67.gif" height="500px" />

## Testing Instructions






This PR can be tested by following these steps:

A lot changed here. Every card that has thumbnails on it should be able to be expanded/collapsed. Checking every issue will be tedious 😕 so sorry. The best way to do this is trigger the error on one page and then duplicate that page at least 5 times.

Errors that will display a button *if* there are more than 5 pages with the issue.

|Issue|How to trigger
|--|--|
|Element with link too big|Add an element to a page with a link and make the element very big|
|Element with link too small|Add an element to a page with a link and make the element very small|
|Image element missing alt|Add an image and remove the description under the `Accessibility` panel|
|Image element resolution|Add an image with very low resolution or a very small image from your computer (less than 828px x 1792px|
|Text contrast too low with page background|Add text that has low color contrast with the page background|
|Page too little text|Don't add any text to the page|
|Page too many links|Add more than 3 elements with links to a page|
|Page too much text|Add more than 200 characters to a page|
|Text font size too small|Add text with a font size less than 12px|
|Video too long|Add a video longer than 60 seconds|
|Video no captions|Add a video without captions (most videos don't have captions)|
|Video no description|Add a video and remove the description text under the `Accessibility` panel|
|Video no poster|Remove the poster image for the video|
|Video element resolution too low|Add a video that is less than 852px x 480px|

Also, when optimizing multiple videos that will also show the collapse/expand button in the checklist.

All other issues should not show the collapse/expand button.

## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #9350
